packages/code-du-travail-frontend/src/outils/DureePreavisLicenciement/steps/Status.js

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import Link from "next/link";
+import data from "@cdt/data...preavis-licenciement/data.json";
+import { Toast } from "@socialgouv/react-ui";
+import { YesNoQuestion } from "../../common/YesNoQuestion";
+import { SelectQuestion } from "../../common/SelectQuestion";
+import {
+  getOptions,
+  getNextQuestionKey,
+  filterSituations,
+  getSituationsFor
+} from "../../common/situations.utils";
+
+const { questions, situations: allSituations } = data;
+const questionsMap = questions.reduce(
+  (state, { name, question }) => ({ ...state, [name]: question }),
+  {}
+);
+
+function validate({ seriousMisconduct }) {
+  const errors = {};
+  if (seriousMisconduct) {
+    errors.seriousMisconduct = (
+      <Toast variant="warning">
+        Dans le cas d’un licenciement pour faute grave ou lourde, il n’y pas
+        d’obligation de respecter un préavis.Vous pouvez trouver plus
+        d’informations sur le préavis de licenciement sur{" "}
+        <Link
+          href="/fiche-service-public/[slug]"
+          as={`/fiche-service-public/preavis-de-licenciement`}
+        >
+          <a>cette fiche</a>
+        </Link>
+        .
+      </Toast>
+    );
+  }
+  return errors;
+}
+
+function StepStatus({ form }) {
+  const { seriousMisconduct, disabledWorker } = form.getState().values;
+  const initialSituations = getSituationsFor(allSituations, { idcc: "0000" });
+
+  const possibleSituations = filterSituations(initialSituations, {});
+  const nextQuestionKey = getNextQuestionKey(possibleSituations, {});
+  const nextQuestionOptions = getOptions(possibleSituations, nextQuestionKey);
```

Comment:
this seems risquy to me to rely on situations to find the questions especially when you know wich next question it should be. also next question algorythm may change and is data driven so it may also change depending on data

Reply:
Yes it's ok for me